### PR TITLE
Enhancements suggestions

### PR DESCRIPTION
**Pull request by xakraz**
For dev:
* Dockerfile
* docker-compose

Features:
* S3 signed URL support (for private file access)
* Proxified download (if using authentication with the HTTP server)

### CODE REVIEW

**Comment by xakraz** [comment]
@powdahound any thoughts ? If you don't have time, maybe we can just close it :)

File: www/index.php [hunk 1]
```diff
@@ -23,11 +23,14 @@
...
 }
 
-
+// Instanciate a S3Browser
 $s3b = new S3Browser($config['bucket-name'], $config['s3-access-key'],
                      $config['s3-secret-key'], $config['s3-ssl'], $config['s3-endpoint']);
 $s3b->enableCaching($config['cache-dir'], $config['cache-time']);
 
+// Set the default timezone
+date_default_timezone_set('Europe/Paris');
```
Comment (powdahound) [inline www/index.php]: This would best be moved to a config value to keep it accurate for all users

File: www/themes/plain/index.tpl.php [hunk 2]
```diff
@@ -163,11 +171,11 @@
     <? endforeach; ?>
     </ul>
     <? endif; ?>
-  
+
   </div>
-  
+
   <div id="footer">
-    <p>Powered by <a href="http://github.com/powdahound/s3browser/" target="_blank">S3 Browser</a></p>
```
Comment (powdahound) [inline www/themes/plain/index.tpl.php]: Let's keep this pointing at the canonical repo :)

File: www/include/S3Browser.php [hunk 3]
```diff
@@ -138,6 +142,14 @@ public function getTree() {
           $cur[$part]['hsize'] = self::formatSize($data['size']);
           $cur[$part]['path'] = $cur[$part]['name'];
           $cur[$part]['name'] = $part;
+          // AWS s3v4 signature
+          $cur[$part]['signature'] = parse_url(
```
Comment (powdahound) [inline www/include/S3Browser.php]: Does this cause any issues if the object is public?

File: README.md [hunk 4]
```diff
@@ -3,31 +3,15 @@ Amazon S3 Directory Browser
 
 Please report any issues [here on GitHub](https://github.com/powdahound/s3browser/issues).
 
-Installation on Heroku (preferred)
```
Comment (powdahound) [inline README.md]: Why remove the Heroku setup? I know folks have enjoyed the ease of this in the past, though this project doesn't see much usage these days.

**Review by powdahound — CHANGES_REQUESTED** [review]
Thanks for this. Left a few comments and questions for you

**Comment by powdahound** [comment]
This is pretty ancient now. If anyone wants any of these bits feel free to pull into a new PR.

**Pull request closed without merging**